webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PolicyWriterService.scala

```diff
@@ -795,6 +801,49 @@ class PolicyWriterServiceImpl(
 
   ///////////// utilities /////////////
 
+  // check if we can use an atomic move or if we need a standard one
+  def getMoveOptions(examplePath: NodePoliciesPaths): IOResult[File.CopyOptions] = {
+    val default    = StandardCopyOption.REPLACE_EXISTING :: Nil
+    val atomically = StandardCopyOption.ATOMIC_MOVE :: default
+
+    def testMove(file: File, destination: String): Task[File] = {
+      IO.effect {
+        val dest = File(destination).parent
+        if(file.parent.path == dest.path) file // same directory is ok
+        else {
+          dest.createDirectoryIfNotExists(true)
+          file.moveTo(dest)(atomically)
+        }
+      }.catchAll(ex => IO.effect(file.delete(true)) *> ex.fail)
+    }
+
+    // try to create a file in new folder parent, move it to base folder, move
+    // it to backup. In case of AtomicNotSupported execption, revert to
+    // simple move
+    (for {
+      n  <- currentTimeNanos
+      f1 <- IO.effect {
+              File(examplePath.newFolder).parent.createChild("test-rudder-policy-mv-options" + n.toString, true, true)
+            }
+      f2 <- testMove(f1, examplePath.baseFolder)
+      _  <- PolicyLoggerPure.trace(s"Can use atomic move from policy new folder to base folder")
+      f3 <- testMove(f2, examplePath.backupFolder)
+      _  <- PolicyLoggerPure.trace(s"Can use atomic move from policy base folder to archive folder")
+      _  <- PolicyLoggerPure.debug(s"Using atomic move for node policies")
+      _  <- IO.effect(f3.delete(true))
+    } yield {
+      atomically
+    }).catchAll {
+      case ex: AtomicMoveNotSupportedException =>
+        PolicyLoggerPure.warn(s"Node policy directories (${examplePath.baseFolder}, ${examplePath.newFolder}, ${examplePath.backupFolder}) " +
+                              s"are not on the same file system. Rudder won't be able to use atomic move for policies, which may have dire " +
+                              s"effect on policy generation performance. You should use the file system.") *>
+        default.succeed
+      case ex: Throwable => SystemError("Error when testing for policy 'mv' mode", ex).fail
```

Comment:
root exception is missing in my error
[2019-12-18 04:57:13] ERROR policy.generation - Error when updating policy, reason was: Cannot write nodes configuration <- Error when testing for policy 'mv' mode
[2019-12-18 04:57:13] ERROR policy.generation - Root exception was: /var/rudder/share/de0727bb-b594-4929-bebd-2ca258c47204/rules.new/test-rudder-policy-mv-options8612784980732448 -> /var/rudder/share/de0727bb-b594-4929-bebd-2ca258c47204/rules

Reply:
it's the box displaying that is not nice. It say that it can't move to dest, but we miss the reason. Most likely, the problem is because parent doesn't exists and I didn't created the correct level.

Reply:
I will change the exception reporting too

Reply:
I don't see what the problem is, I added logs - I fear we will need to test it again to understand.

Reply:
I added logs and it was because I forgot the filename in the move, so it was trying to overwrite a directory with file :/. Now corrected. Perf are not better